lib/hammer_cli_foreman/compute_resource/utils.rb

```diff
@@ -8,6 +8,14 @@ def self.get_image_uuid(compute_resource_id, image_id)
       )
     end
 
+    def self.get_hostgroup_compute_resource_id(hostgroup_id)
+      HammerCLIForeman.record_to_common_format(
+          HammerCLIForeman.foreman_resource(:hostgroups).call(
+              :show, 'id' => hostgroup_id
+          )['compute_resource_id']
```

Comment:
After some testing I've noticed, that if we use host group that is nested then the server response contains only the name of a compute resource (`"compute_resource_id" => nil`), so this will cause error `Missing argument for 'compute_resource'`.

What do you think will be better:
  1. Support resource id resolving in case we have the resource name only?
  2. Fix this on API side.

Reply:
@ofedoren good catch! thanks @ofedoren.
I'm working on a fix for that in foreman-core. I will update this one as soon as I finish with the PR .
 thanks